DMDisassembler/Program.cs

```diff
@@ -71,32 +76,116 @@ static void Main(string[] args) {
+                break;
+            }
+            default: {
+                Console.WriteLine($"No additional help for \"{command}\"");
+                AllCommands();
+                break;
+            }
+        }
+
+        void AllCommands()
+        {
+            Console.WriteLine("DM Disassembler for OpenDream");
+            Console.WriteLine("Commands and arguments:");
+            Console.WriteLine("help [command]            : Show additional help for [command] if applicable");
+            Console.WriteLine("quit|q                    : Exits the disassembler");
+            Console.WriteLine("search type|proc [name]   : Search for a particular typepath or a proc on a selected type");
+            Console.WriteLine("select|sel                : Select a typepath to run further commands on");
+            Console.WriteLine("list procs|globals        : List all globals, or all procs on a selected type");
+            Console.WriteLine("decompile|d [name]        : Decompiles the proc on the selected type");
+            Console.WriteLine("stats [type]              : Prints various stats about the game. Use \"help stats\" for more info");
+            Console.WriteLine("dump-all                  : Decompiles every proc and writes the output to a file");
+            Console.WriteLine("test-all                  : Tries to decompile every single proc to check for issues with this disassembler; not for production use");
+        }
+    }
+
+    private static void Stats([CanBeNull] string statType) {
+        if (string.IsNullOrEmpty(statType)) {
+            PrintHelp("stats");
+            return;
+        }
+
+        switch (statType) {
+            case "procs-by-type": {
+                ProcsByType();
+                return;
+            }
+            default: {
+                Console.WriteLine($"Unknown stat \"{statType}\"");
+                PrintHelp("stats");
+                return;
+            }
+        }
+
+        void ProcsByType() {
+            Console.WriteLine("Counting all proc declarations (no overrides) by type. This may take a moment.");
+            Dictionary<int, int> typeIdToProcCount = new Dictionary<int, int>();
+            foreach (DMProc proc in Procs) {
+                if(proc.IsOverride || proc.Name == "<init>") continue; // Don't count overrides or <init> procs
+                if (typeIdToProcCount.TryGetValue(proc.OwningTypeId, out var count)) {
+                    typeIdToProcCount[proc.OwningTypeId] = count + 1;
+                } else {
+                    typeIdToProcCount[proc.OwningTypeId] = 1;
+                }
+            }
+
+            var sorted = typeIdToProcCount.OrderByDescending(kvp => kvp.Value).ToList();
+            Console.WriteLine("Type: Proc Declarations");
+            for (int i = 0; i < sorted.Count; i++) {
+                var pair = sorted[i];
```

Comment:
You can avoid the large & slow list creation
```suggestion
            Console.WriteLine("Type: Proc Declarations");
            foreach (var pair in typeIdToProcCount.OrderByDescending(kvp => kvp.Value)) {
```